### PR TITLE
Qdma mailbox and ERT support

### PR DESCRIPTION
**Pull request by houlz0507**

### CODE REVIEW

File: src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c [hunk 1]
```diff
@@ -483,9 +487,15 @@ static int xclmgmt_setup_msix(struct xclmgmt_dev *lro)
 	 * cover its own usr vectors. So, only the last chunk of the table will
 	 * ever be used for mgmt pf.
 	 */
-	lro->msix_user_start_vector = XOCL_READ_REG32(lro->core.intr_bar_addr +
-		XCLMGMT_INTR_USER_VECTOR) & 0x0f;
-	total = lro->msix_user_start_vector + XCLMGMT_MAX_USER_INTR;
+	if (dev_info->flags & XOCL_DSAFLAG_FIXED_INTR) {
```
Comment (maxzhen) [inline src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c]: The comment above does not apply to fixed intr case. please move it into the else clause.

File: src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c [hunk 2]
```diff
@@ -863,6 +878,8 @@ static int xclmgmt_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 		return rc;
 	}
 
+	pci_set_master(pdev);
```
Comment (maxzhen) [inline src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c]: Why do we need to set it explicitly now? Does it change anything for XDMA?

File: src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c (hunk 1 above)
Comment (houlz0507) [reply src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c]: done

File: src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c (hunk 2 above)
Comment (houlz0507) [reply src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c]: Bus Master is required before configuring MSIX

**Pull request merged**